tests/integration_test.py

```diff
@@ -1073,6 +1082,64 @@ def runTest(self):
         self.assertTrue(sock.fileno() > -1)
 
 
+class TestRunContainerReadingSocket(BaseTestCase):
```

Comment:
Yeh, this is hard to test isn't it! While I'm not a fan of copying in the dockertpy code, I appreciate it's difficult to write a test without it.

Minor point: please don't abbreviate variable names like `nxt` `nxt_size`, keep it fully expanded and descriptive, `next` `next_size`, readability is important. Thanks.